clang/lib/CIR/CodeGen/CIRGenOpenMPRuntime.cpp

```diff
@@ -52,3 +52,37 @@ bool CIRGenOpenMPRuntime::emitTargetGlobal(clang::GlobalDecl &GD) {
   assert(!UnimplementedFeature::openMPRuntime());
   return false;
 }
+
+void CIRGenOpenMPRuntime::emitTaskWaitCall(CIRGenBuilderTy &builder,
+                                           CIRGenFunction &CGF,
+                                           mlir::Location Loc,
+                                           const OMPTaskDataTy &Data) {
+  if (CGF.CGM.getLangOpts().OpenMPIRBuilder && Data.Dependences.empty()) {
+    // TODO: Need to support taskwait with dependences in the OpenMPIRBuilder.
+    // TODO(cir): This could change in the near future when OpenMP 5.0 gets
+    // supported by MLIR
+    builder.create<mlir::omp::TaskwaitOp>(Loc);
+  } else {
+    llvm_unreachable("NYI");
+  }
+}
+
+void CIRGenOpenMPRuntime::emitBarrierCall(CIRGenBuilderTy &builder,
+                                          CIRGenFunction &CGF,
+                                          mlir::Location Loc) {
+  if (CGF.CGM.getLangOpts().OpenMPIRBuilder) {
```

Comment:
Still work to be done. Please follow the skeleton, you should either add unrecheables or `assert(!Unimplemented::...)`, here's another example for you:

```
void CGOpenMPRuntime::emitBarrierCall(...) {
  // Check if we should use the OMPBuilder
  assert(!Unimplemented::openMPRegionInfo());
  if (CGF.CGM.getLangOpts().OpenMPIRBuilder) {
    builder.create<mlir::omp::BarrierOp>(Loc);
    return;
  }

  if (!CGF.HaveInsertPoint())
    return;

  llvm_unreacheable("NYI");
}
```

Please review the other methods and make sure you abide to this approach.